tests/physics/resolve/ball_ball/test_ball_ball.py

```diff
@@ -111,3 +111,82 @@ def test_head_on_z_spin(model: BallBallCollisionStrategy, cb_wz_i: float):
     cb_wz_f = cb_f.state.rvw[2][2]
     assert cb_wz_f > 0, "Spin direction shouldn't reverse"
     assert cb_wz_f < cb_wz_i, "Spin should be decay"
+
+
+@pytest.mark.parametrize(
+    "model", [FrictionalInelastic(), FrictionalMathavan(num_iterations=int(1e5))]
+)
+@pytest.mark.parametrize("speed", np.logspace(-1, 1, 5))
+@pytest.mark.parametrize("cut_angle_radians", np.linspace(0, math.pi / 2.0, 10))
+def test_gearing_z_spin(
+    model: BallBallCollisionStrategy, speed: float, cut_angle_radians: float
+):
+    """Cue ball has positive z-spin (e.g. hitting right-hand-side of cue ball)"""
+    cb_i, ob_i = ball_collision()
+
+    # velocity
+    cb_i.state.rvw[1] = speed * np.array(
+        [math.cos(cut_angle_radians), math.sin(cut_angle_radians), 0.0]
+    )
+    # angular velocity
+    cb_i.state.rvw[2] = np.array(
+        [0.0, 0.0, -cb_i.vel[1] / cb_i.params.R]
+    )  # Gearing side-spin
+
+    v_c = ptmath.surface_velocity(
+        cb_i.state.rvw, np.array([1.0, 0.0, 0.0]), cb_i.params.R
+    ) - np.array([cb_i.vel[0], 0.0, 0.0])
+    assert ptmath.norm3d(v_c) < 1e-10, "Relative surface contact speed should be zero"
+
+    cb_f, ob_f = model.resolve(cb_i, ob_i, inplace=False)
+
+    assert abs(ob_f.vel[1]) < 1e-3, "Gearing english shouldn't cause throw"
+    assert abs(ob_f.avel[2]) < 1e-3, "Gearing english shouldn't cause induced side-spin"
+
+
+@pytest.mark.parametrize("model", [FrictionalInelastic()])
```

Comment:
This test is pretty specific to FrictionalInelastic. Mathavan doesn't really work for this one even with 10e5 iterations, nor would I expect it to.